client/src/partials/templates/bhDateEditor.tmpl.html

```diff
@@ -0,0 +1,37 @@
+<!-- Date editor component view implementation-->
+<div 
+	class="form-group"	
+	ng-class="{'has-error' : $ctrl.validationTrigger && $ctrl.form.dateEditorValue.$invalid}" >
+
+	<!-- Input component implementation-->
+	<div class="input-group">
+	    <input
+	    	data-date-editor-input
+	    	name="dateEditorValue"
+	    	type="text"
+	    	class="form-control"	    	
+	    	uib-datepicker-popup="{{$ctrl.dateFormat}}"
+	    	ng-model="$ctrl.dateValue"
+	    	is-open="$ctrl.editMode"
+	    	min-date="$ctrl.minDate"
+	    	max-date="$ctrl.maxDate"
+	    	close-text="Close"
+	    	ng-disabled="!$ctrl.editMode"
+	    	show-button-bar="true"
+	    	required
+	    	/>
+	    <span class="input-group-btn">
+	    	<button id="set-date-btn" class="btn btn-default" type="button" ng-click="$ctrl.changeMode()" ng-disabled="$ctrl.editMode">
```

Comment:
Do you ever use this id?  Since the HTML spec only allows for unique `id` elements on a page, could you replace this with a `data-*` tag?